### ISSUE TITLE
Move range

### ISSUE THREAD

**Issue by freeit** [andmarti1424/sc-im#711]
How can I move a source range to a destination range like the original sc command "rm"?  Especially with its bevaviour:

"... This differs from deleting a range with rx and pulling it back in with pm in that any expressions that  reference a cell in the range to be moved will reference the cell at its new address after the move."

**Comment by andmarti1424** [comment]
Have you checked doc/help?
This might help:

![image](https://user-images.githubusercontent.com/9466161/165599524-8e777968-dc42-41cd-9366-772e9d33a3e0.png)

**Comment by freeit** [comment]
Yes I have checked the docs, but didn't find a corresponding command. If you think the "Pc" command is doing the right thing, I think you're wrong, because this command adjust the cell references from formulars inside the range of cells not the references from cells outside the range to be copied. Let me show it in an example:

Following spreadsheet:
A0=10
A1=A0

Now if you move cell content A0 to B0 with the "rm" command from "sc" the spreadsheet looks like this:
B0=10
A1=B0

How would you do this with "sc-im"?

**Comment by andmarti1424** [comment]
Oh yes. Now I get what you're trying to do. We might have missed adding that during clean up.

**Comment by freeit** [comment]
It would be great, if you could add this feature/command again, because it's absolutely necessary if you want to rearrange your sheet.

By the way, I like your new "sc" (sc-im) very much. Really good work.

**Comment by andmarti1424** [comment]
Yep. Actually I was checking out how much trouble would be..

**Comment by andmarti1424** [comment]
[711.patch.txt](https://github.com/andmarti1424/sc-im/files/8578608/711.patch.txt)
Please feel free to try this patch over dev branch.

**Comment by freeit** [comment]
Your patch works for references in plain cells. It doesn't work for references for example in functions (see attached video; it is  showing how it works correctly in sc). Another point is the non-atomic operation. In sc the rm command is an atomic operation (like shown in the attached video). In the patched sc-im it is not. You have to do the range move in two separate operations (dd and Pm). After the dd op

https://user-images.githubusercontent.com/526117/165732470-69e040fb-e8ee-4cc9-b2cb-c149ce2d87f5.mp4

eration, the spreadsheet may be in an erroneous state, because of missing reference endpoints. Only after the Pm operation the spreadsheet (maybe?) change back to an orderly status.

**Comment by andmarti1424** [comment]
Will check out the references within functions. I'm aware of the two steps needed for the movement. It's that way on purpose. It should not stay in erroneous state between the two steps, cause the references are not lost when you yank the original range.
